nullplatform/dimension.go

```diff
@@ -82,12 +108,39 @@ func (c *NullClient) GetDimension(dimensionID string) (*Dimension, error) {
 		return nil, fmt.Errorf("error getting dimension resource, got status code: %d, body: %s", res.StatusCode, string(body))
 	}
 
-	dimension := &Dimension{}
-	err = json.Unmarshal(body, dimension)
+	response := &map[string]interface{}{}
+	err = json.Unmarshal(body, response)
 	if err != nil {
 		return nil, fmt.Errorf("error decoding dimension: %v", err)
 	}
 
+	results, ok := (*response)["results"].([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("the data has returnned no occurence")
+	}
+
+	// Check if "results" has exactly one element
+	if len(results) != 1 {
+		return nil, fmt.Errorf("result expected returned %d elements", len(results))
+	}
+
+	dimension := &Dimension{}
+	rawDimension := results[0].(map[string]interface{})
+	dimension.ID = int(rawDimension["id"].(float64))
+	dimension.Name = rawDimension["name"].(string)
+	dimension.Status = rawDimension["status"].(string)
+	dimension.NRN = rawDimension["nrn"].(string)
+	dimension.Order = int(rawDimension["order"].(float64))
+	dimension.Slug = rawDimension["slug"].(string)
+	values := rawDimension["values"].([]interface{})
+	dimension.Values = make([]map[string]interface{}, len(values))
+	for i, v := range values {
+		dimension.Values[i] = v.(map[string]interface{})
```

Comment:
idem any

Reply:
y handleo de assertion (up to you aca)

Reply:
done el any. El assertion, sobre cual? el casteo de la 138 o el make?